File: javalin-http2-example/src/main/java/app/JavalinHttp2ExampleApp.java
package app;

import io.javalin.Javalin;
import io.javalin.community.ssl.SslPlugin;
import io.javalin.http.staticfiles.Location;

public class JavalinHttp2ExampleApp {

    public static void main(String[] args) {

        Javalin app = Javalin.create(config -> {

            // Set up the SSL plugin (Enables HTTP2 by default)
            config.registerPlugin(new SslPlugin(ssl -> {
                ssl.keystoreFromClasspath("keystore.jks", "password"); // The ssl plugin will enable HTTP/2 by default
                ssl.insecurePort = 8080;
                ssl.securePort = 8443;
            }));

            config.staticFiles.add("/public", Location.CLASSPATH);

            config.routes.get("/", ctx -> ctx.result("Hello World"));
        }).start();

        System.out.println("Server started with HTTP/2 support!");
        System.out.println("  HTTP/1.1: http://localhost:8080/static-files-test.html");
        System.out.println("  HTTP/2:   https://localhost:8443/static-files-test.html");
        System.out.println("Compare the network tab in your browser to see HTTP/2 in action");
        System.out.println("Notice the order of execution/number of simultaneous requests");

    }

}

File: javalin-http2-example/src/main/kotlin/app/JavalinHttp2ExampleApp.kt
package app

import io.javalin.Javalin
import io.javalin.community.ssl.SslPlugin
import io.javalin.http.staticfiles.Location

fun main() {

    Javalin.create { javalinConfig ->
        // Set up the SSL plugin (Enables HTTP2 by default)
        javalinConfig.registerPlugin(SslPlugin { ssl ->
            ssl.keystoreFromClasspath("keystore.jks", "password")
            ssl.insecurePort = 8080
            ssl.securePort = 8443
        })
        javalinConfig.staticFiles.add("/public", Location.CLASSPATH)

        javalinConfig.routes.get("/") { it.result("Hello World") }
    }.start()

    println("Server started with HTTP/2 support!")
    println("  HTTP/1.1: http://localhost:8080/static-files-test.html")
    println("  HTTP/2:   https://localhost:8443/static-files-test.html")
    println("Compare the network tab in your browser to see HTTP/2 in action")
    println("Notice the order of execution/number of simultaneous requests")

}

